client/src/test/java/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationsServiceTest.java

```diff
@@ -0,0 +1,288 @@
+
+    when(api.conversationListRecentConversations(
+            eq(uriPartID), eq("conversation app Id"), eq(true), eq(1), eq(null), eq("ASC")))
+        .thenReturn(conversationsRecentListPage0);
+
+    when(api.conversationListRecentConversations(
+            eq(uriPartID), eq("conversation app Id"), eq(true), eq(1), eq("1"), eq("ASC")))
+        .thenReturn(conversationsRecentListPage1);
+
+    ConversationsListRecentRequest request =
+        ConversationsListRecentRequest.builder()
+            .setAppId("conversation app Id")
+            .setOnlyActive(true)
+            .setPageSize(1)
+            .setOrder(OrderEnum.ASC)
+            .build();
+
+    ConversationsListRecentResponse response0 = service.listRecent(request);
+    Collection<ConversationRecentMessage> content = response0.getContent();
+    Assertions.assertThat(content.size()).isEqualTo(1);
+    TestHelpers.recursiveEquals(
+        content.stream().findFirst().orElse(null),
+        ConversationRecentMessageDtoTest.conversationRecentMessage);
+    Assertions.assertThat(response0.hasNextPage()).isEqualTo(true);
+
+    ConversationsListRecentResponse response1 = response0.nextPage();
+    content = response1.getContent();
+    Assertions.assertThat(content.size()).isEqualTo(1);
+    TestHelpers.recursiveEquals(
+        content.stream().findFirst().orElse(null),
+        ConversationRecentMessageDtoTest.conversationRecentMessage);
+    Assertions.assertThat(response1.hasNextPage()).isEqualTo(false);
+  }
+
+  @Test
+  void stopActive() throws ApiException {
+
+    service.stopActive(ConversationDtoTest.conversation.getId());
+
+    verify(api)
+        .conversationStopActiveConversation(
+            projectIdCaptor.capture(), conversationIdCaptor.capture());
+
+    Assertions.assertThat(projectIdCaptor.getValue()).isEqualTo(uriPartID);
+    Assertions.assertThat(conversationIdCaptor.getValue())
+        .isEqualTo(ConversationDtoTest.conversation.getId());
+  }
+
+  @Test
+  void delete() throws ApiException {
+
+    service.delete(ConversationDtoTest.conversation.getId());
+
+    verify(api)
+        .conversationDeleteConversation(projectIdCaptor.capture(), conversationIdCaptor.capture());
+
+    Assertions.assertThat(projectIdCaptor.getValue()).isEqualTo(uriPartID);
+    Assertions.assertThat(conversationIdCaptor.getValue())
+        .isEqualTo(ConversationDtoTest.conversation.getId());
+  }
+
+  @Test
+  void update() throws ApiException {
+    when(api.conversationUpdateConversation(
+            eq(uriPartID),
+            eq(ConversationDtoTest.conversation.getId()),
+            eq(ConversationDtoTest.conversation),
+            eq(null),
+            eq(MetadataUpdateStrategy.MERGE_PATCH)))
+        .thenReturn(ConversationDtoTest.conversation);
+
+    Conversation response =
+        service.update(
+            ConversationDtoTest.conversation.getId(),
+            MetadataUpdateStrategy.MERGE_PATCH,
+            ConversationDtoTest.conversation);
```

Comment:
Indeed, when updating a conversation, you can add the `id` but it won't be taken into account

Reply:
Yes and unit test are based onto specs definitions: specs says a Conversation instance can be send to server and then with an id even if ignored... 
If specs are refined, this mean unit tests will have to be updated and this is the purpose of such tests 😎